docs/proposal-0004-jetstream-preview.md

```diff
@@ -0,0 +1,118 @@
+* optionally, user generate preview data
+* optionally, users can view preview data on a looker dashboard
+* config is merged and experiment analysis run gets triggered
+
+## Implementation
+
+### Jetstream
+
+Jetstream needs a Python environment and access to BigQuery in order to run successfully. A new CLI command could be added to the existings jetstream CLI:
+
+```
+> jetstream preview
+
+Options:
+--project_id              Project to write preview data to
+--dataset_id              Dataset to write preview data to
+--slug                    Slug of the experiment
+--config_file             Path/URL to custom config file
+--config_repo             Path/URL to repo with metric definitions
+--start_date              Date for which first analysis should be run
+--end_date                Date for which last analsis should be run
+--generate-population     Generate a population sample (useful if enrollment hasn't happened yet)
+--population-sample-size  The generated population sample size
+--sql-output-dir          Path to write executed SQL 
+--platform                Platform/app to run analysis for
+```
+
+By default the preview would run jetstream on a 3 day analysis window on a population sample of 1%.
+These default parameters can be overwritten when invoking `jetstream preview`.
+
+The resulting data artifacts get written into the `mozdata.tmp` dataset by default, which is configured to delete data after 7 days. Anyone at Mozilla has permissions to write to this dataset.
+
+Once jetstream has generated the preview data it will print out a URL to a Looker dashboard. The Looker dashboard allows users to visualize the preview data. 
+
+The dashboard is a static dashboard that has been added to Looker. It has a filter field that will be pre-filled throught the URL pointing to the temporary datasets that have been generated. It will also have filters so users can select the analysis period, analysis basis, segment, metric and statistic they would like to preview. The visualization will be a line graph with confidence intervals that shows a line for every branch and comparison.
+
+The queries that get executed will be saved locally into files which makes it possible for users to debug the queries (enrollment and metrics) that get executed.
+
+Running the preview command will also print out cost estimates for running analyses on the experiment.
+
+The jetstream CI will need to be updated to publish jetstream to pypi. This way users do not need to pull in the jetstream repository, but can simply run `pip install mozilla-jetstream`.
+
+Jetstream fetches information on how experiments are configured from the Experimenter API. In some cases users might want to get a preview for experiments that haven't been launched yet or that haven't seen any enrollments. For these experiments a random population sample can be generated when specifying `--generate-population` that will be considered as the clients that enrolled.
```

Comment:
This is a fun idea, I like this. It would be awesome to be able to test configs before experiments are live